pkg/plugins/git/generic/main.go

```diff
@@ -71,51 +94,54 @@ func Checkout(branch, remoteBranch, workingDir string) error {
 		})
 
 		if err != nil {
+			logrus.Debugf("Branch: '%v' - \n\t%v", branch, err)
 			return err
 		}
 
-		logrus.Debugf("Checkout branch: '%v'", remoteBranch)
 		// Checkout locale branch without creating it yet
 		err = w.Checkout(&git.CheckoutOptions{
 			Branch: plumbing.NewBranchReferenceName(remoteBranch),
-			Create: false,
+			Create: true,
 			Keep:   false,
 			Force:  true,
 		})
 
-		// Branch doesn't exist locally
-		if err == plumbing.ErrReferenceNotFound {
-			logrus.Debugf("Creating first branch: '%v'", remoteBranch)
-
-			err = w.Checkout(&git.CheckoutOptions{
-				Branch: plumbing.NewBranchReferenceName(remoteBranch),
-				Create: true,
-				Keep:   false,
-				Force:  true,
-			})
+		logrus.Debugf("Creating branch '%v'", remoteBranch)
 
-			if err != nil &&
-				(err != plumbing.ErrReferenceNotFound ||
-					err != git.ErrBranchExists) {
-				return err
-			}
-
-		} else if err != nil && err != plumbing.ErrReferenceNotFound {
+		if err != nil {
 			return err
 		}
+
 	} else if err != plumbing.ErrReferenceNotFound && err != nil {
+		logrus.Debugln(err)
 		return err
 	} else {
+
 		// Means that a local branch named remoteBranch already exist
 		// so we want to be sure that the local branch is
 		// aligned with the remote one.
-		remoteBranchRef := fmt.Sprintf("refs/remotes/origin/%s", remoteBranch)
 
-		logrus.Infof(remoteBranchRef)
+		remote, err := r.Remote("origin")
+		if err != nil {
+			return err
+		}
+		refs, err := remote.List(&git.ListOptions{})
```

Comment:
```suggestion
		refs, _ := remote.List(&git.ListOptions{})
```
(or add a `if err!:nil` block after).
Reported by the linter:
```
pkg/plugins/git/generic/main.go:128:9: ineffectual assignment to `err` (ineffassign)
                refs, err := remote.List(&git.ListOptions{})
```

Reply:
Thanks for the catching that :), I'll add a if err !nil block